### PR TITLE
feat: Updated exit documentation

### PR DESCRIPTION
**Pull request by balloonpopper**
Clarifying when to use ESCexit instead of ESCitem.

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/esc_exit.gd [hunk 1]
```diff
@@ -1,4 +1,13 @@
-# An item that streamlines exiting scenes
+# An item that streamlines exiting scenes.
+#
+# Use this when you want this object to only change scenes (with an optional
+# sound effect.)
```
Comment (dploeger) [inline addons/escoria-core/game/core-scripts/esc_exit.gd:4]:
```suggestion
# sound effect).
```

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 2]
```diff
@@ -1,4 +1,7 @@
-# ESCItem is a Sprite that defines an item, potentially interactive
+# ESCItem is a Sprite that defines an item, potentially interactive.
```
Comment (dploeger) [inline addons/escoria-core/game/core-scripts/esc_item.gd:1]: Ah, I just see it's actually not a sprite. That suggests a visual representation. You still need to add an actual Sprite2D to it. We should change that bit as well while you're at it.

I'd suggest:
```suggestion
# ESCItem defines a potentially interactive item of the game.
```

**Review by dploeger — CHANGES_REQUESTED** [review]
Awesome. Some small remarks.

File: addons/escoria-core/game/core-scripts/esc_item.gd (hunk 2 above)
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc_item.gd:1]:
```suggestion
ESCItem defines a potentially interactive item in the game.
```

**Comment by balloonpopper** [comment]
Updated @dploeger

File: addons/escoria-core/game/core-scripts/esc_exit.gd [hunk 3]
```diff
@@ -1,4 +1,13 @@
-# An item that streamlines exiting scenes
+# An item that streamlines exiting scenes.
```
Comment (balloonpopper) [inline addons/escoria-core/game/core-scripts/esc_exit.gd:1]:
```suggestion
# An ESCExit is an item that streamlines the process to exit a scene.
```

**Review by dploeger — APPROVED** [review]
Wonderful. Thanks. 🎁

File: addons/escoria-core/game/core-scripts/esc_exit.gd [hunk 4]
```diff
@@ -1,4 +1,15 @@
-# An item that streamlines exiting scenes
+# An ESCExit is a minimal feature node that provides an exit to a room.
+#
+# For exits that don't require scripts, the ``ESCExit`` node is provided. 
+# The only things you will need to configure on the node are the 
+# "target_scene" to change to, and optionally, a "switch 
+# sound" (the sound to play when changing rooms).
+#
+# If you want to attach the exit to a script to perform additional actions -
+# a cutscene for example, use an ESCItem with "Is Exit" selected instead.
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_exit.gd:9]:
```suggestion
# If you want to attach the exit to a script to perform additional actions -
# a cutscene for example - use an ``ESCItem`` with "Is Exit" selected instead.
```

File: addons/escoria-core/game/core-scripts/esc_exit.gd [hunk 5]
```diff
@@ -1,4 +1,15 @@
...
+#
+# The game character will automatically walk to an ESCLocation created as a
+# child of an ESCExit node.
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_exit.gd:12]:
```suggestion
# The game character will automatically walk to an ``ESCLocation`` created as a
# child of an ``ESCExit`` node.
```

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 6]
```diff
@@ -1,4 +1,14 @@
...
+#
+# When interacting with an ESCItem, the game character will automatically
+# walk to an ESCLocation that is created as a child of an ESCItem.
+#
+# By selecting the "Is Exit" checkbox when you create an ``ESCItem``
+# node, Escoria will look for a ":exit_scene" event in the attached script file.
+# Any commands you place in the ":exit_scene" event will be run when the player
+# chooses to "use" the exit - for example, saying a goodbye, or running a
+# cutscene. Place a "change_scene" command as part of this event to move the
+# character to the next room.
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_item.gd:11]:
```suggestion
# An ``ESCItem`` defines a (usually interactive) item in the game.
#
# When interacting with an ``ESCItem``, the game character will automatically
# walk to an ``ESCLocation`` that is created as a child of an ``ESCItem``.
#
# By selecting the "Is Exit" checkbox when you create an ``ESCItem``
# node, Escoria will look for an ``:exit_scene`` event in the attached script file.
# Any commands you place in the ``:exit_scene`` event will be run when the player
# chooses to "use" the exit - for example, saying a goodbye, or running a
# cutscene. Place a ``change_scene`` command inside this event to move the
# character to the next room.
```

File: addons/escoria-core/game/core-scripts/esc_item.gd [hunk 7]
```diff
@@ -53,7 +63,9 @@ export(String) var global_id
 # The ESC script for this item
 export(String, FILE, "*.esc") var esc_script
 
-# If true, the ESC script may have an ":exit_scene" event to manage scene changes
+# If true, the ESC script may have an ":exit_scene" event to manage scene changes.
+# For simple exits that do not require scripted actions, the ESCExit node may be
```
Comment (BHSDuncan) [inline addons/escoria-core/game/core-scripts/esc_item.gd:67]:
```suggestion
# If true, the ESC script may have an ``:exit_scene`` event to manage scene changes.
# For simple exits that do not require scripted actions, the ``ESCExit`` node may be
```

**Review by BHSDuncan — CHANGES_REQUESTED** [review]

**Review by StraToN — APPROVED** [review]

**Review by BHSDuncan — APPROVED** [review]

**Pull request merged**